### ISSUE TITLE
Images are empty

### ISSUE THREAD

**Issue by sergei-grechanik** [Thiago4532/mdmath.nvim#3]
Hi, I'm trying to test the plugin, and I see that images are uploaded, and they have the correct size, but they are completely empty:
![image](https://github.com/user-attachments/assets/0e7b45d9-2bec-44f1-9f2d-b4f0f0ab2819)
Same in st (bounding boxes are shown, one of the images is previewed in feh):
![image](https://github.com/user-attachments/assets/d1d34f28-8944-451f-b446-278e72f56b8f)

To be honest I don't know where to start debugging this.

Here is my config:
```
  mdmath.setup({
    -- Filetypes that the plugin will be enabled by default.
    filetypes = {},
    -- Color of the equation, can be a highlight group or a hex color.
    -- Examples: 'Normal', '#ff0000'
    foreground = '#ff0000',
    -- Hide the text when the equation is under the cursor.
    anticonceal = true,
    -- Hide the text when in the Insert Mode.
    hide_on_insert = true,
    -- Scale of the equation images, increase to prevent blurry images when increasing terminal
    -- font, high values may produce aliased images.
    scale = 2.0,
  })
```

**Comment by Thiago4532** [comment]
Don’t worry about debugging it yourself—this code’s a bit of a mess right now, haha! I’m in the middle of refactoring and working on documentation to make PRs easier. I’ll check out your issue later today.

**Comment by Thiago4532** [comment]
What system are you running? Also what are your NeoVim and ImageMagick versions? And are you using some terminal multiplexer?

**Comment by sergei-grechanik** [comment]
Ubuntu 24.04

```
$ convert --version
Version: ImageMagick 6.9.12-98 Q16 x86_64 18038
```
(I tried it also with ImageMagick7 on my PATH, but it's still the same)

```
$ nvim --version
NVIM v0.11.0-dev-108+g22fe04452
Build type: RelWithDebInfo
LuaJIT 2.1.1716656478
```
(I also tried nvim v0.10.2 from snap, same thing)

No multiplexers

**Comment by Thiago4532** [comment]
Okay, so apparently it's related to how ImageMagick delegates the SVG. I've discovered that my ImageMagick is using librsvg to convert the SVG, I've tried using Inkscape and got a problem that is similar to yours.

I read online that ImageMagick on Ubuntu doesn't come with RSVG support, not sure if this is still true, but this may be why it's not working in Ubuntu.

That is news to me, I didn't know ImageMagick delegated to another program, and for me this is unreliable, since different backends lead to different results.

Tomorrow I'll try to use librsvg directly, or try to make it work using only Inkscape, but I'll probably move out from ImageMagick to convert the SVG, and use it only for other image processing tasks.
